kitsune/src/service/post.rs

```diff
@@ -207,16 +303,20 @@ impl PostService {
             subject.clean_html();
             subject
         });
+
         let mut content = if create_post.process_markdown {
-            let parser = Parser::new_ext(&create_post.content, Options::all());
-            let mut buf = String::new();
-            html::push_html(&mut buf, parser);
-            buf
+            process_markdown(&create_post.content)
         } else {
             create_post.content
         };
         content.clean_html();
-        let content_lang = kitsune_language::detect_language(DetectionBackend::default(), &content);
+
+        let detect_language =
+            |s: &str| kitsune_language::detect_language(DetectionBackend::default(), s);
+        let content_lang = create_post.language.map_or_else(
+            || detect_language(&content),
```

Comment:
I think you can just inline the closure here. The path `kitsune_language::detect_language` makes it pretty clear what it does here

Reply:
Just another small ping because of this.  
Would you prefer to keep it inside a separate closure? Because I personally don't see an advantage

Reply:
The thing is that here it's used twice, so I kept it in. I only removed it from `update` because there it was only called once so there wasn't a point, but I think here it makes things a bit more concise.

Reply:
Oh, I completely overlooked the second usage. You're right.

(been a while since I was last on my laptop, damn stuff is so much smaller)